syntax = "proto2";

package janitor;

enum Mode {
    // Don't make any changes to this package
    skip = 0;

    // Push changes directly to the packaging repository
    push = 1;

    // Propose the changes in a pull request
    propose = 2;

    // Attempt to push the changes, but if that fails, create a pull request
    // instead.
    attempt_push = 3;

    // Build the package, but don't propose/push the changes.
    build_only = 4;
}

enum ChangelogMode {
    auto = 0;

    update_changelog = 1;

    leave_changelog = 2;
}

message Match {
    // Maintainer e-mail
    repeated string maintainer = 1;

    // Uploader e-mail
    repeated string uploader = 2;

    // Source package name
    repeated string source_package = 3;
}

message Suite {
    optional string name = 1;

    optional string command = 2;

    optional Mode mode = 3;
};

message Policy {
    repeated Match match = 1;

    repeated Suite suite = 2;

    // Changelog update mode
    optional ChangelogMode changelog = 3;
}

message PolicyConfig {
    repeated Policy policy = 1;
}
